sse2neon.h

```diff
@@ -8873,13 +8873,48 @@ static inline int _sse2neon_sido_negative(int res, int lb, int imm8, int bound)
 }
 
 #define SSE2NEON_MIN(x, y) (x) < (y) ? (x) : (y)
-#define SSE2NEON_GET_LENGTH_OR_BOUND(la, lb, bound) \
-    int tmp1 = la ^ (la >> 31);                     \
-    la = tmp1 - (la >> 31);                         \
-    int tmp2 = lb ^ (lb >> 31);                     \
-    lb = tmp2 - (lb >> 31);                         \
-    la = SSE2NEON_MIN(la, bound);                   \
-    lb = SSE2NEON_MIN(lb, bound);
+
+#define SSE2NEON_CMPESTRX_GET_UPPER_BOUND \
+    const int bound = (imm8 & 0x01) ? 8 : 16;
+
+#define SSE2NEON_CMPESTRX_COMPARE_THEN_AGGREGATE(la, lb, imm8)     \
+    SSE2NEON_CMPESTRX_GET_UPPER_BOUND                              \
+    int tmp1 = la ^ (la >> 31);                                    \
+    la = tmp1 - (la >> 31);                                        \
+    int tmp2 = lb ^ (lb >> 31);                                    \
+    lb = tmp2 - (lb >> 31);                                        \
+    la = SSE2NEON_MIN(la, bound);                                  \
+    lb = SSE2NEON_MIN(lb, bound);                                  \
+    int r2 = (_sse2neon_cmpfunc_table[imm8 & 0x0f])(a, la, b, lb); \
+    r2 = _sse2neon_sido_negative(r2, lb, imm8, bound);
+
+// Compare packed strings in a and b with lengths la and lb using the control
+// in imm8, and returns 1 if b did not contain a null character and the
+// resulting mask was zero, and 0 otherwise.
+// https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_cmpestra
+FORCE_INLINE int _mm_cmpestra(__m128i a,
+                              int la,
+                              __m128i b,
+                              int lb,
+                              const int imm8)
+{
+    int lb_cpy = lb;
+    SSE2NEON_CMPESTRX_COMPARE_THEN_AGGREGATE(la, lb, imm8)
```

Comment:
It looks weird that there is no ending semicolon (`;`) at the end of this macro.